regparser/tree/xml_parser/reg_text.py

```diff
@@ -73,13 +73,30 @@ def build_tree(reg_xml):
 
     part = reg_xml.xpath('//PART')[0]
 
+    # Build a list of SUBPARTs, then pull SUBJGRPs into that list:
     subpart_xmls = [c for c in part.getchildren() if c.tag == 'SUBPART']
+    subpart_and_subjgrp_xmls = []
     if len(subpart_xmls) > 0:
-        subparts = [build_subpart(reg_part, s) for s in subpart_xmls]
-        tree.children = subparts
+        for subpart in subpart_xmls:
+            subpart_and_subjgrp_xmls.append(subpart)
+            subjgrps = [c for c in subpart.getchildren() if c.tag == 'SUBJGRP']
+            for subjgrp in subjgrps:
+                subpart_and_subjgrp_xmls.append(subjgrp)
+
+    if len(subpart_and_subjgrp_xmls) > 0:
+        subthings = []
+        letter_list = []
+        for subthing in subpart_and_subjgrp_xmls:
+            if subthing.tag == "SUBPART":
+                subthings.append(build_subpart(reg_part, subthing))
+            elif subthing.tag == "SUBJGRP":
+                letter_list, built_subjgrp = build_subjgrp(
+                    reg_part, subthing, letter_list)
```

Comment:
I think the letter_list concatenation could be bumped out to here. I think:
```python
subjgrp = build_subgrp(reg_part, subthing, letter_list)
letter_list.append(subjgrp.label[-1])
```

Random other suggestion: would it make sense for the letter_list to be a set?

And one more: Should the letter_list include the subpart letters? To prevent collisions?